### PR TITLE
Mineral floors!

### PR DESCRIPTION
**Pull request by unid15**
Working and with no conflicts this time

Adds the ability to create floor tiles from all minerals
Floors from common minerals (plasma, uranium, gold, silver, diamond) have no special effect (other than diamond floors not being damaged by IEDs)
Floors from bananium honk when touched and squeak when stepped on
Floors from phazon improve your walking speed, but are easily broken by IEDs

http://a.pomf.se/tbeuxx.webm

### CODE REVIEW

**Comment by FudgePucker** [comment]
Uranium irradiating the area around it when ever walked on would be an interesting feature.

**Comment by DrCelt** [comment]
I second @FudgePucker 's suggestion

File: code/game/turfs/simulated/floor.dm [hunk 1]
```diff
@@ -276,6 +293,15 @@ turf/simulated/floor/proc/update_icon()
 	else if(is_grass_floor())
 		src.icon_state = "sand[pick("1","2","3")]"
 		broken = 1
+	else if(is_mineral_floor())
+		if(material=="diamond") return //diamond doesn't break
+		if(material=="phazon") //Phazon shatters
```
Comment (Iamgoofball) [inline code/game/turfs/simulated/floor.dm]: 1. why aren't these just using the material datum for the material
2. why are these snowflaked in and why aren't they procs on said material datum

**Comment by Kurfursten** [comment]
@FudgePucker @DrCelt It needs a more terrifying looking sprite if that's the case. Right now it looks like green carpet -- which is fine -- but if it's going to be radiated, a new player needs to be able to look at it and say right away "that is dangerous". Consider animated glowing.

**Comment by DrLeprikon** [comment]
Could the silver floors clean themselves the way they do on that dwarf mode station?

**Comment by AutismSpider** [comment]
silver floor should  clean themselves after bit of time

gold should give off bit of light like pda levels of light.

**Comment by AutismSpider** [comment]
No plastic floors?

**Comment by FudgePucker** [comment]
There aren't even any plastic walls.

**Comment by AutismSpider** [comment]
We lack  wood  and plastic walls though.

**Comment by Unfit** [comment]
Okay so uranium floors now emit radiation and glo when stepped on. It's not a large-scale chain reaction (like uranium walls), and the radiation emitted isn't very significant (but it adds up quickly).
You can also add reinforced glass to uranium tiles to make them safe to walk on.



I'm not a big fan of other ideas, so I'd say that'd be all for now. 
Webm of the whole thing soon

**Comment by FudgePucker** [comment]
If you do decide to add plastic floors, they need some effect to make them worth it. Maybe make them slip proof, for when you are really pissed off at the clown who got ahold of lube.

**Comment by FudgePucker** [comment]
Wait does the glass covered uranium floors still light up when walked on?
I want to make sure my dance club is as awesome as possible.

**Comment by Unfit** [comment]
Nope it doesn't

**Comment by clusterfack** [comment]
This is neat, it also has conflicts

**Comment by Unfit** [comment]
i hate github so much

**Comment by clusterfack** [comment]
Also time for squash, ask on IRC a bit later if you don't know how

**Comment by Unfit** [comment]
oh wow I think i fixed it

**Comment by Unfit** [comment]
no conflicts anymore

**Comment by Unfit** [comment]
@clusterfack no more conflicts remove that label please

**Comment by ComicIronic** [comment]
Did someone say no more conflicts

File: code/game/objects/items/devices/aicard.dm [hunk 2]
```diff
@@ -102,10 +102,6 @@
 						for(var/mob/living/silicon/ai/A in src)
 							A.suiciding = 1
 							A << "Your core files are being wiped!"
-							A.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has been wiped with an [src.name] by [U.name] ([U.ckey])</font>")
```
Comment (ComicIronic) [inline code/game/objects/items/devices/aicard.dm]: Why would you do this

File: code/game/turfs/simulated/mineral.dm [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+		if(floor_tile)
+			material = floor_tile.material
+		..()
+
+/turf/simulated/floor/mineral/plasma
+	name = "plasma floor"
+	icon_state = "plasma"
+	floor_tile = /obj/item/stack/tile/mineral/plasma
+
+	New()
```
Comment (ComicIronic) [inline code/game/turfs/simulated/mineral.dm]: Absolute pathing why

**Pull request merged**